website/docs/guides/managed_service_identity.html.markdown

```diff
@@ -71,12 +71,20 @@ Terraform can be configured to use managed identity for authentication in one of
 
 ### Configuring with environment variables
 
-Setting the `ARM_USE_MSI` environment variable to `true` tells Terraform to use a managed identity. In addition to a properly-configured management identity, Terraform needs to know the subscription ID and tenant ID to identify the full context for the Azure provider.
+Setting the `ARM_USE_MSI` environment variable to `true` tells Terraform to use a managed identity.
+
+By default, Terraform will use the system assigned identity for authentication. If users want to use user assigned identity instead, the `ARM_CLIENT_ID` have to be specified to the [client id](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/user_assigned_identity#client_id) of the identity.
+
+By default, Terraform will use the MSI endpoint provided by MSI VM Extension to get the authentication token, which covers the most use cases. Whilst the endpoint might be different (e.g. Azure Function App) in other cases, where users need to explicitly specify the endpoint via `ARM_MSI_ENDPOINT`.
```

Comment:
this is no longer a VM Extension as far as I'm aware, more something the hypervisor exposes? but same here, could we highlight the provider block field (as above)

Reply:
I changed it to "well-known" endpoint.